daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/GrpcLedgerClient.scala

```diff
@@ -71,6 +75,17 @@ class GrpcLedgerClient(val grpcClient: LedgerClient, val applicationId: Applicat
     TransactionFilter(parties.toList.map(p => (p, filters)).toMap)
   }
 
+  // Template filter with the package id removed, for upgrades
+  private def upgradeableTemplateFilter(
+      parties: OneAnd[Set, Ref.Party],
+      templateId: Identifier,
+  ): TransactionFilter = {
+    val filters = Filters(
+      Some(InclusiveFilters(Seq(toApiIdentifier(templateId).copy(packageId = ""))))
```

Comment:
does this do what we expect (i.e. match template ids ignoring the package id)? [this](https://github.com/DACH-NY/canton/blob/2e551c667eb4d4ce65850190c6593858758a73ed/community/ledger/ledger-api-core/src/main/scala/com/digitalasset/canton/ledger/api/domain/domain.scala#L29-L32) was the relevant code I found in canton, but it seems to use `Identifier` equality (through `.contains`)

Reply:
I've tested that it worked locally, but I didn't verify that it ensures the name is the same. I'll check that

Reply:
It does what we expect, yes. It only returns contracts with the matching name. Also, we have another check in the ValueTranslator that ensures the name matches for a valid upgrade